collectors/channel_collector_test.go

```diff
@@ -0,0 +1,84 @@
+package collectors
+
+import (
+	"testing"
+
+	"github.com/btcsuite/btcutil"
+	"github.com/lightningnetwork/lnd/lnrpc"
+	"github.com/stretchr/testify/require"
+)
+
+type inboundFeeTestCase struct {
+	name              string
+	amt               btcutil.Amount
+	expectedFee       btcutil.Amount
+	expectNoLiquidity bool
+}
+
+var (
+	remotePolicies = map[uint64]*lnrpc.RoutingPolicy{
+		1: {
+			FeeBaseMsat:      20000,
+			FeeRateMilliMsat: 10000,
+		},
+		2: {
+			FeeBaseMsat:      250000,
+			FeeRateMilliMsat: 6000,
+		},
+	}
+	remoteBalances = map[uint64]btcutil.Amount{
+		1: 10000,
+		2: 10000,
+	}
+
+	testCases = []inboundFeeTestCase{
+		{
+			name:        "single channel use all",
+			amt:         10000,
+			expectedFee: 120,
+		},
+		{
+			name:        "single channel partially used",
+			amt:         5000,
+			expectedFee: 70,
+		},
+		{
+			name:              "not enough",
+			amt:               25000,
+			expectNoLiquidity: true,
+		},
+		{
+			name:        "two channels use all",
+			amt:         20000,
+			expectedFee: 120 + 310,
+		},
+		{
+			name:        "two channels partially used",
+			amt:         15000,
+			expectedFee: 120 + 280,
+		},
+	}
+)
+
+// TestGetInboundFee tests the specific-fee based inbound fee calculation.
+func TestGetInboundFee(t *testing.T) {
+	for _, test := range testCases {
+		test := test
+
+		t.Run(test.name, func(t *testing.T) {
+			testGetInboundFee(t, test)
+		})
+	}
+}
+
+func testGetInboundFee(t *testing.T, test inboundFeeTestCase) {
+	fee := getInboundFee(test.amt, remotePolicies, remoteBalances)
+
+	if test.expectNoLiquidity {
+		require.Nil(t, fee, "expected no liquidity")
```

Comment:
cool, never noticed you could add a message to `require`

Reply:
yes, with parameters even if you use `require.Nilf`